extensions/functions_datetime.yaml

```
@@ -265,3 +342,111 @@ scalar_functions:
           - name: y
             value: interval_year
         return: boolean
+  -
+    name: "assume_timezone"
+    description: Convert local time to UTC using given local time's timezone.
+    impls:
+      - args:
+          - name: x
+            value: timestamp
+          - name: timezone
+            value: string
+        return: timestamp_tz
+  -
+    name: "strptime"
+    description: Parse string into timestamp/date/time using provided format,
+      see https://man7.org/linux/man-pages/man3/strptime.3.html for reference.
+    impls:
+      - args:
+          - name: timestamp_string
+            value: string
+          - name: format
+            value: string
+          - name: timezone
+            value: string
+        return: timestamp_tz
+      - args:
+          - name: timestamp_string
+            value: string
+          - name: format
+            value: string
+        return: timestamp
+  -
+    name: "strftime"
+    description: Convert timestamp/date/time to string using provided format,
+      see https://man7.org/linux/man-pages/man3/strftime.3.html for reference.
+    impls:
+      - args:
+          - name: x
+            value: timestamp
+          - name: format
+            value: string
+        return: string
+      - args:
+          - name: x
+            value: timestamp_tz
+          - name: format
+            value: string
+          - name: timezone
+            value: string
+        return: string
+      - args:
+          - name: x
+            value: date
+          - name: format
+            value: string
+        return: string
+      - args:
+          - name: x
+            value: time
+          - name: format
+            value: string
+        return: string
+  -
+    name: "round_temporal"
+    description: Floor a given timestamp to a multiple of a time unit. Time unit can be physical or calendar based.
```

```suggestion
    description: Round a given timestamp to a multiple of a time unit. Time unit can be physical or calendar based.
```

since the direction can be chosen using an option.

Also, this is a function with lots of annoying special cases that need to be specified. For example, what happens if you round to a multiple of 11 milliseconds? None of the time scales have 11 as a divisor, and time being relative by definition, there is no unambiguous implicit epoch to round to. Likewise, the behavior for 0 and negative multiples must be specified, which I guess just means error out.